packages/ember-runtime/lib/computed/computed_macros.js

```diff
@@ -20,7 +20,10 @@ function expandPropertiesToArray(properties) {
   }
 
   for (let i = 0; i < properties.length; i++) {
-    expandProperties(properties[i], extractProperty);
+    let property = properties[i];
+    assert('Dependent keys passed to Ember.computed.and() or Ember.computed.or() can\'t have spaces.', property.indexOf(' ') < 0);
```

Comment:
@rwjblue @courajs I wonder if the only way to  know which macro is being processed is to pass it along explicitly (ie. "and" or "or"), since the predicate from the caller `generateComputedWithPredicate` is a callback that can be arbitrary?